windows/src/developer/kmcmpdll/Compiler.cpp

```diff
@@ -612,147 +611,151 @@ DWORD ValidateMatchNomatchOutput(PWSTR p) {
-		if(!q) return CERR_InvalidLanguageName;
+    q = GetDelimitedString(&p, L"\"\"", 0);
+    if (!q) return CERR_InvalidLanguageName;
 
-    if((msg = AddStore(fk, TSS_LANGUAGENAME, q)) != CERR_None) return msg;
-		break;
+    if ((msg = AddStore(fk, TSS_LANGUAGENAME, q)) != CERR_None) return msg;
+    break;
 
-	case T_LANGUAGE:
-    {
-      WarnDeprecatedHeader();   // I4866
-      wchar_t *tokcontext = NULL;
-		  q = wcstok_s(p, L"\n", &tokcontext);  // I3481
-		  if((msg = AddStore(fk, TSS_LANGUAGE, q)) != CERR_None) return msg;
-		  break;
-    }
-	case T_LAYOUT:
-    {
-      WarnDeprecatedHeader();   // I4866
-      wchar_t *tokcontext = NULL;
-  		q = wcstok_s(p, L"\n", &tokcontext);  // I3481
-	  	if((msg = AddStore(fk, TSS_LAYOUT, q)) != CERR_None) return msg;
-		  break;
-    }
-	case T_CAPSOFF:
+  case T_LANGUAGE:
+  {
+    WarnDeprecatedHeader();   // I4866
+    wchar_t *tokcontext = NULL;
+    q = wcstok_s(p, L"\n", &tokcontext);  // I3481
+    if ((msg = AddStore(fk, TSS_LANGUAGE, q)) != CERR_None) return msg;
+    break;
+  }
+  case T_LAYOUT:
+  {
+    WarnDeprecatedHeader();   // I4866
+    wchar_t *tokcontext = NULL;
+    q = wcstok_s(p, L"\n", &tokcontext);  // I3481
+    if ((msg = AddStore(fk, TSS_LAYOUT, q)) != CERR_None) return msg;
+    break;
+  }
+  case T_CAPSOFF:
     WarnDeprecatedHeader();   // I4866
-		if((msg = AddStore(fk, TSS_CAPSALWAYSOFF, L"1")) != CERR_None) return msg;
-		break;
+    if ((msg = AddStore(fk, TSS_CAPSALWAYSOFF, L"1")) != CERR_None) return msg;
+    break;
 
-	case T_CAPSON:
+  case T_CAPSON:
     WarnDeprecatedHeader();   // I4866
-		if((msg = AddStore(fk, TSS_CAPSONONLY, L"1")) != CERR_None) return msg;
-		break;
+    if ((msg = AddStore(fk, TSS_CAPSONONLY, L"1")) != CERR_None) return msg;
+    break;
 
-	case T_SHIFT:
+  case T_SHIFT:
     WarnDeprecatedHeader();   // I4866
-		if((msg = AddStore(fk, TSS_SHIFTFREESCAPS, L"1")) != CERR_None) return msg;
-		break;
+    if ((msg = AddStore(fk, TSS_SHIFTFREESCAPS, L"1")) != CERR_None) return msg;
+    break;
 
-	case T_HOTKEY:
-    {
-      WarnDeprecatedHeader();   // I4866
-      wchar_t *tokcontext = NULL;
-      if((q = wcstok_s(p, L"\n", &tokcontext)) == NULL) return CERR_CodeInvalidInThisSection;  // I3481
-	  	if((msg = AddStore(fk, TSS_HOTKEY, q)) != CERR_None) return msg;
-		  break;
+  case T_HOTKEY:
+  {
+    WarnDeprecatedHeader();   // I4866
+    wchar_t *tokcontext = NULL;
+    if ((q = wcstok_s(p, L"\n", &tokcontext)) == NULL) return CERR_CodeInvalidInThisSection;  // I3481
+    if ((msg = AddStore(fk, TSS_HOTKEY, q)) != CERR_None) return msg;
+    break;
+  }
+  case T_BITMAP:
+  {
+    WarnDeprecatedHeader();   // I4866
+    wchar_t *tokcontext = NULL;
+    if ((q = wcstok_s(p, L"\n", &tokcontext)) == NULL) return CERR_InvalidBitmapLine;  // I3481
+
+    while (iswspace(*q)) q++;
+    if (*q == '"') {
+      p = q;
+      q = GetDelimitedString(&p, L"\"\"", 0);
+      if (!q) return CERR_InvalidBitmapLine;
```

Comment:
I see this line - the `if (!q)` part - seems to be new.  Given its position within the code... this indicates when an error occurred within `GetDelimitedString`?

Reply:
Minor post-review note - I don't think I saw any edits to `GetDelimitedString` within the PR.

Reply:
`GetDelimitedString` can return `NULL` with invalid input. I think in this particular case it couldn't actually happen, because other tests prevent bad outcomes, but we should have been testing the result like we do elsewhere.